trusted-types/block-string-assignment-to-Element-setAttributeNode.html

```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<meta http-equiv="Content-Security-Policy" content="require-trusted-types-for 'script';">
+<title>trusted-types (TT): setAttributeNode with node from non-TT realm</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+</head>
+<body>
+<div id="nonSVGTestElements">
+    <iframe srcdoc="v"></iframe>
+    <embed src="v">
+    <script src="v"></script>
+    <object data="v"></object>
+    <object codebase="v"></object>
+</div>
+<svg id="svgTestElements">
+    <script href="v"></script>
+    <script xlink:href="v"></script>
+</svg>
+<script>
+    const passThroughPolicy = trustedTypes.createPolicy("passThrough", { createHTML: s => s });
+
+    function runTest(aTestElement) {
+        const testAttr = aTestElement.attributes[0];
+
+        async_test(t => {
+            const sourceFrame = document.createElement("iframe");
+
+            sourceFrame.srcdoc = passThroughPolicy.createHTML(
+            `<!DOCTYPE html>
+            <head>
+            <meta charset="utf-8">
+            </head>
+            <body><div ` + testAttr.name + `="` + testAttr.value + `"></div>
+            doc without TT CSP.
+            </body>`);
+
+            t.add_cleanup(() => {
+                sourceFrame.remove();
+            });
+
+            sourceFrame.addEventListener("load", t.step_func_done(() => {
+                // A window is a global object which has 1-to-1 mapping to a realm, see the first
+                // note of <https://html.spec.whatwg.org/#realms-settings-objects-global-objects>
+                // and its following paragraph.
+                assert_not_equals(aTestElement.ownerDocument, null);
+                assert_not_equals(aTestElement.ownerDocument, undefined);
+                assert_not_equals(aTestElement.ownerDocument.defaultView, sourceFrame.contentWindow,
```

Comment:
Again, assert_equals would be better. But this is also not really testing anything as you haven't done anything with aTestElement so this is always going to be what you expect except in implementations that were already broken, no?

Reply:
How would this look like with `assert_equals`?

Yes, this is always going to be what's expected. It's just documentation expressed in code. Documentation itself can be wrong.

Reply:
You'd pass `self` as a second argument? But I'm still not really sure how these asserts are helpful as you're not doing anything with `aTestElement`. It would make sense to assert these things for `sourceAttr` before and after, maybe.

Reply:
@annevk makes sense, changed it. PTAL again.